src/hypothesis/statistics.py

```diff
@@ -62,6 +62,8 @@ def __init__(self, engine):
             self.exit_reason = 'nothing left to do'
         elif engine.exit_reason == ExitReason.flaky:
             self.exit_reason = 'test was flaky'
+        elif engine.exit_reason == ExitReason.max_shrinks:
+            self.exit_reason = 'shrunk example %s times' % (MAX_SHRINKS,)
```

Comment:
Off topic, possibly: Do you like this large elif? It seems like we could eliminate it by using the same design as Exception. Each reason would be an instance of a subtype of Reason, with a message attribute. Then we don't have this action-at-a-distance problem keeping these messages in sync with their far-away implementation.